### ISSUE TITLE
Bug in MobileViTForSemanticSegmentation output shape

### ISSUE THREAD

**Issue by kevinpl07** [huggingface/transformers#20255]
### System Info

Python 3.9.6
transformers==4.24.0

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
from transformers import MobileViTFeatureExtractor, MobileViTForSemanticSegmentation
from PIL import Image
import requests

url = "http://images.cocodataset.org/val2017/000000039769.jpg"
image = Image.open(requests.get(url, stream=True).raw)

feature_extractor = MobileViTFeatureExtractor.from_pretrained("apple/deeplabv3-mobilevit-small")
model = MobileViTForSemanticSegmentation.from_pretrained("apple/deeplabv3-mobilevit-small")

inputs = feature_extractor(images=image, return_tensors="pt")

outputs = model(**inputs)
logits = outputs.logits
predicted_mask = logits.argmax(1).squeeze(0)
```

### Expected behavior

When running inference on pretrained MobileViTForSemanticSegmentation as in [here](https://huggingface.co/apple/deeplabv3-mobilevit-small), I get very bad segmentation mask in the huggingface online inference. When running locally I see that the output mask has a shape of (1, 21, 32, 32).

I don't think outputting a 32x32 mask is intended or am I wrong?

**Comment by NielsRogge** [comment]
Hi,

MobileViT's semantic segmentation is indeed very low-resolution, as seen in this Space: https://huggingface.co/spaces/Matthijs/mobilevit-deeplab-demo. cc @hollance 

If you want to look into models that output more fine-grained segmentation results, I'd recommend checking out [SegFormer](https://huggingface.co/docs/transformers/model_doc/segformer) and [MaskFormer](https://huggingface.co/docs/transformers/model_doc/maskformer).

**Comment by hollance** [comment]
The original model will do a bilinear upsampling to 513x513 which makes the mask larger but also more blurry. In the Transformers library we don't do this upsampling in the model as we don't know ahead of time how you want to use the results. That's why the Transformers model outputs the low resolution version.

**Comment by NielsRogge** [comment]
Closing this issue as it has been resolved, feel free to reopen.
